en/ui-system/components/engine/ui-material.md

```diff
@@ -0,0 +1,47 @@
+# UI Custom Material
+We supports UI custom material in sprite component since v1.2, the user interface is as follows:
+
+![](ui-material/UIMaterial.png)
+
+Its usage is no different from other materials, but because the SpriteComponent has some function based on the UI-built-in materials, there are some points to note:
+
+- When the number of custom materials is set to 0 or empty, the default material will be used. Please refer to [SpriteComponent](../editor/sprite.md)
+- UI does not support multiple materials, the number of custom materials is at most one
+- When the ui custom material is used, the Grayscale function on the panel will be invalid. Users can choose to implement this function in the material
+- After setting the custom material, the Color property on the panel will be mixed with the color set in the material itself (if had) to affect the rendering color of the sprite
+- For custom materials, you need to introduce the cc-sprite-texture header file in the shader and sample the cc_spriteTexture, you can directly set the image resources used by the sprite through the SpriteFrame on the settings panel
```

Comment:
For custom materials, you need to introduce the cc-sprite-texture header file in the shader and sample the cc_spriteTexture, you can directly set the image resources used by the sprite through the SpriteFrame on the settings panel
-->
For custom materials, developers need to introduce the `cc-sprite-texture` header file in the shader and sample the `cc_spriteTexture`. Image resources can be directly set and used by the sprite through the `SpriteFrame` on the settings panel.